internal/peer/redis.go

```diff
@@ -139,15 +139,24 @@ func (p *redisPeers) RegisterUpdatedPeersCallback(cb func()) {
 func (p *redisPeers) registerSelf() {
 	tk := time.NewTicker(refreshCacheInterval)
 	for range tk.C {
-		// every 5 seconds, insert a 30sec timeout record
-		p.store.Register(context.TODO(), p.publicAddr, peerEntryTimeout)
+		ctx, cancel := context.WithTimeout(context.Background(), p.c.GetPeerTimeout())
+		// every 5 seconds, insert a 30sec timeout record. we ignore the error
+		// here since Register() logs the error for us.
+		p.store.Register(ctx, p.publicAddr, peerEntryTimeout)
+		cancel()
 	}
 }
 
 func (p *redisPeers) updatePeerListOnce() {
-	currentPeers, err := p.store.GetMembers(context.TODO())
+	ctx, cancel := context.WithTimeout(context.Background(), p.c.GetPeerTimeout())
+	defer cancel()
+
+	currentPeers, err := p.store.GetMembers(ctx)
 	if err != nil {
-		// TODO maybe do something better here?
+		logrus.WithField("name", p.publicAddr).
```

Comment:
A nit, but could you please do this with logrus.WithFields rather than a succession of WithField calls?

Reply:
@thrawn01 -- to clarify, although I approved this PR, I would like to see that change before we merge it. Are you able to do that, please?

Reply:
Done.